model/model_builder.py

```diff
@@ -178,6 +183,49 @@ def add_output_layer(self):
 
         return self
 
+    def add_secondary_output(self):
```

Comment:
Might be, one problem is that we don't want to overwrite the training op for either could maybe be dealt with using a boolean